metavoice-1b/model/model.py

```diff
@@ -0,0 +1,147 @@
+from pathlib import Path
+import tempfile
+from typing import Any, Literal, Optional, Tuple
+from dataclasses import dataclass
+import torch
+from huggingface_hub import snapshot_download
+import io
+import base64
+from scipy.io import wavfile
+from fam.llm.enhancers import get_enhancer
+import os
+
+from fam.llm.sample import (
+    InferenceConfig,
+    Model,
+    build_models,
+    get_first_stage_path,
+    get_second_stage_path,
+    sample_utterance,
+)
+
+
+HF_MODEL_ID = "metavoiceio/metavoice-1B-v0.1"
+
+@dataclass
+class ServingConfig:
+    max_new_tokens: int = 864 * 2
+    """Maximum number of new tokens to generate from the first stage model."""
+
+    temperature: float = 1.0
+    """Temperature for sampling applied to both models."""
+
+    top_k: int = 200
+    """Top k for sampling applied to both models."""
+
+    seed: int = 1337
+    """Random seed for sampling."""
+
+    dtype: Literal["bfloat16", "float16", "float32", "tfloat32"] = "bfloat16"
+    """Data type to use for sampling."""
+
+    enhancer: Optional[Literal["df"]] = "df"
+    """Enhancer to use for post-processing."""
+
+@dataclass
+class TTSRequest:
+    text: str
+    guidance: Optional[Tuple[float, float]] = (3.0, 1.0)
+    top_p: Optional[float] = 0.95
+    speaker_ref_path: Optional[str] = None
+    top_k: Optional[int] = None
+
+@dataclass
+class ModelState:
+    """Data class for holding model state."""
+    spkemb_model: Optional[torch.nn.Module]
+    first_stage_model: Optional[Model]
+    second_stage_model: Optional[Model]
+    enhancer: Optional[object]
+
+class Model:
+    def __init__(self, **kwargs):
+        self._secrets = kwargs["secrets"]
+        self.config = ServingConfig()
```

Comment:
could we call this `DefaultInferenceParams` or something like that instead? I feel like there are too many things called `config` right now (the `InferenceConfig`, `config.yaml`, and this)